### DISCUSSION TITLE
关于 Gemini 1.5 文件上传

### DISCUSSION THREAD

**Discussion by sxjeru** [lobehub/lobehub#2151] [Development Question] [answered]
尝试过直接修改本模型 `files: true`，发现一些问题：
- 聊天窗口目前只能预览图片；
- vercel 部署仅允许上传 4.5 MB 以下文件；

本人非计算机专业，但有意愿贡献本项目的 Gemini 部分，因此想在此提供一下开发方向。

1. Gemini 1.5 官方支持多模态 image (max 3600 files) 与 audio，私下测试也支持 txt，不支持 pdf/doc/xls 等。所以至少添加音乐与文本文件的预览，并考虑对不同模型作上传文件格式限制。
2. 根据[文档](https://ai.google.dev/gemini-api/docs/prompting_with_media#upload_a_file_to_the_file_api)，是否认为这种上传可以规避 vercel 等的限制，当然仅限 Gemini 模型。

这边基本不敢改动程序结构，寄希望于开发者能有所想法，并搭建一个框架，对于具体的细枝末节，咱也能稍微帮忙完善。

最后感谢开发此优秀项目！

**Comment by arvinxx** [comment, marked as answer]
@sxjeru 下个月会做文件上传的整体方案，会给出一个框架性的构建方案（类似之前做 Model Provider 这样），然后到时候 gemini 部分的实现欢迎一起共建~
